app/Http/Controllers/Admin/AdminController.php

```diff
@@ -18,6 +18,7 @@ public function __construct()
     public function index()
     {
         $users = request()->has('search') ? SearchUsers::get(request('search')) : User::findAllPaginated();
+        $users->appends(request()->except('page'));
```

Comment:
Let's indeed append this inside the `SearchUsers` object. Just append the search result, not the entire request. Do this for the `SearchThreads` as well.